code/string_calculator/src/string_calculator.py

```diff
@@ -0,0 +1,66 @@
+import regex as re
+
+
+class StrCalculatorType(object):
+    # Data structure for storing the parsed string
+    # Abstraction for StrCalculator
+    # Let's get rid of a difficult input string,
+    # we will keep only the important details.
+    def __init__(self, string, delimiters, numstring, numbers):
+        self.string = string
+        self.delimiters = delimiters
+        self.numstring = numstring
+        self.numbers = numbers
+
+
+class StrCalculatorParser(object):
+    # string to StrCalculatorType converter
+    DEFAULT_DELIMS = ["\n", ","]
+    CUSTOM_DELIM_REGEX = "^([^\d])\n"
+
+    def parse(self, string):
+        m = self._check_custom_delim(string)
+        if m:
```

Comment:
Can be shortened with `:=` operator with Python 3.8 =)

In any case, `m` is not the best variable name

Reply:
Done.